### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by shrey27**
Added Features

- Sign In page
- Sign up page
- Log out button
- encoded token and user data handling

### CODE REVIEW

File: src/frontend/common/deals/Deal.js [hunk 1]
```diff
@@ -0,0 +1,151 @@
+import './deals.css';
+// import { useCartCtx, useCartAPICtx } from '../../context/cartContext';
+// import { useWishlistCtx } from '../../context/wishlistContext';
```
Comment (manish591) [inline src/frontend/common/deals/Deal.js:3]: Remove commented code if not needed

File: src/frontend/common/header/Category.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+import { PRODUCTS } from '../../../routes';
+
+const categoryList = [
+  'Protein',
+  'Gym Gear',
+  'Athletics',
+  'Nutrition',
+  'Vegan',
+  'Clothing'
+];
```
Comment (manish591) [inline src/frontend/common/header/Category.js:12]: Don't add data here above the component. Add it in a separate file and then import.

**Review by manish591 — CHANGES_REQUESTED** [review]
Code is good but, here are a few suggestions: 
* Remove commented code if not needed. Don't commit this much amount of commented code. 
* I saw a problem in your barrel export using index.js. Inside your common folder, for each subfolder, you're having an index.js file, but you can have just one for the common folder and export all the components from there.

File: src/frontend/common/header/Category.js (hunk 2 above)
Comment (shrey27) [reply src/frontend/common/header/Category.js:12]: this will be removed in final PR

File: src/frontend/authentication/Signin.js [hunk 3]
```diff
@@ -0,0 +1,98 @@
+import './authentication.css';
+import { useEffect } from 'react';
+import Navbar from '../common/navbar';
+import Footer from '../common/footer';
+import { useAuthCtx } from '../context/authenticationContext';
+import { Link } from 'react-router-dom';
+import { HOMEPAGE, SIGNUP } from '../../routes';
```
Comment (anuj-kumary) [inline src/frontend/authentication/Signin.js:7]: Follow one naming convention either use capital or small for import

File: src/frontend/common/deals/Deal.js [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+  // const [addedToCart, setAddedToCart] = useState(false);
+
+  // useEffect(() => {
+  //   if (addedPID && addedPID.includes(pid)) setAddedToWishlist(true);
+  //   else setAddedToWishlist(false);
+
+  //   if (addedCartPID && addedCartPID.includes(pid)) setAddedToCart(true);
+  //   else setAddedToCart(false);
+  // }, [addedCartPID, addedPID, pid]);
+
```
Comment (anuj-kumary) [inline src/frontend/common/deals/Deal.js:40]: Remove this unwanted comment

File: src/frontend/common/header/Category.js (hunk 2 above)
Comment (manish591) [reply src/frontend/common/header/Category.js:12]: cool

**Pull request merged**